### PR TITLE
Add additional stratos safeguards

### PR DESCRIPTION
**Pull request by ismirlia**
This adds safegaurds for:
- volume action (replicationEnabled)
- volume create-v1 (replicationEnabled)
- volume create-v2 (replicationEnabled)
- volume list (auxiliary and replicationEnabled)
- instance update (profileID)

### CODE REVIEW

**Comment by ismirlia** [comment]
From what I could tell:
volume action (replicationEnabled)
volume create-v2 (replicationEnabled)
volume list (auxiliary and replicationEnabled)
are not accessible endpoints through terraform and
instance update (profileID)
is not update-able through terraform

File: clients/instance/ibm-pi-volume.go [hunk 1]
```diff
@@ -215,6 +228,10 @@ func (f *IBMPIVolumeClient) UpdateVolumeAttach(id, volumeID string, body *models
 
 // Performs action on volume
 func (f *IBMPIVolumeClient) VolumeAction(id string, body *models.VolumeAction) error {
+	// Check for satellite differences in this endpoint
+	if strings.Contains(f.session.Options.Zone, helpers.PIStratosRegionPrefix) && body.ReplicationEnabled != nil {
```
Comment (yussufsh) [inline clients/instance/ibm-pi-volume.go:232]: one small comment: Could we have a common function `isStratos()` which can be used everywhere?
```suggestion
	if body.ReplicationEnabled != nil && isStratos() {
```

File: clients/instance/ibm-pi-volume.go [hunk 2]
```diff
@@ -44,6 +45,10 @@ func (f *IBMPIVolumeClient) GetAll() (*models.Volumes, error) {
 	params := p_cloud_volumes.NewPcloudCloudinstancesVolumesGetallParams().
 		WithContext(f.ctx).WithTimeout(helpers.PIGetTimeOut).
 		WithCloudInstanceID(f.cloudInstanceID)
+	// Check for satellite differences in this endpoint
+	if strings.Contains(f.session.Options.Zone, helpers.PIStratosRegionPrefix) && (params.Auxiliary != nil && params.ReplicationEnabled != nil) {
+		return nil, fmt.Errorf("replication enabled parameter is not supported in satellite location, check documentation")
```
Comment (yussufsh) [inline clients/instance/ibm-pi-volume.go:50]: do we also need to add `auxiliary` param as well in the error message?

Comment (ismirlia) [reply clients/instance/ibm-pi-volume.go]: Yes, that is a typo. Made changes.

File: clients/instance/ibm-pi-volume.go (hunk 1 above)
Comment (ismirlia) [reply clients/instance/ibm-pi-volume.go]: Made changes.

File: clients/instance/ibm-pi-cloud-connection.go [hunk 3]
```diff
@@ -27,7 +26,7 @@ func NewIBMPICloudConnectionClient(ctx context.Context, sess *ibmpisession.IBMPI
 
 // Create a Cloud Connection
 func (f *IBMPICloudConnectionClient) Create(body *models.CloudConnectionCreate) (*models.CloudConnection, *models.CloudConnectionCreateResponse, error) {
-	if strings.Contains(f.session.Options.Zone, helpers.PIStratosRegionPrefix) {
+	if ibmpisession.IsOnPrem(f.session.Options.Zone) {
```
Comment (yussufsh) [inline clients/instance/ibm-pi-cloud-connection.go:29]: Don't you think we should add the func in session struct? This way you do not need to pass the zone and can be called as `f.session.IsOnPrem()`.

In session.go:
```
func (s *IBMPISession) IsOnPrem() bool {
  return strings.Contains(s.Options.Zone, helpers.PIStratosRegionPrefix)
}
```

Comment (ismirlia) [reply clients/instance/ibm-pi-cloud-connection.go:29]: I'm not familiar enough with these objects to say. If you think that's more appropriate, I can do that.

**Review by yussufsh — APPROVED** [review]
lgtm

**Pull request merged**